### PR TITLE
COMMERCE-10173 - It's possible to save a product configuration with wrong values

### PR DESCRIPTION
**Pull request by victorcdp**
Related Issue: [COMMERCE-10173](https://issues.liferay.com/browse/COMMERCE-10173)

Problem was that there was no validation of existence in the inputs. Proposed fix is to add a required validator so that the product can't be saved with the specified inputs empty. Also there was no validation at the backend, so this solution includes that as well.

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
@andrea-ale-sbarra , resent with change included from: https://github.com/brianikim/liferay-portal/pull/69

**Comment by andrea-ale-sbarra** [comment]
@victorcdp please fix conflicts here :)

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by victorcdp** [comment]
@andrea-ale-sbarra should be fine now. Thank you.

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
@victorcdp 

com.liferay.commerce.test.CommerceCheckoutTest > testLowStockActivityOnCheckout FAILED because it's misconfiguring those fields

`if (cpDefinitionInventory == null) {
			_cpDefinitionInventoryLocalService.addCPDefinitionInventory(
				_user.getUserId(), cpDefinition.getCPDefinitionId(), "default",
				"default", false, false, 1, false, 0,
				CPDefinitionInventoryConstants.DEFAULT_MAX_ORDER_QUANTITY, null,
				1);
		}
		else {
			_cpDefinitionInventoryLocalService.updateCPDefinitionInventory(
				cpDefinitionInventory.getCPDefinitionInventoryId(), "default",
				"default", false, false, 1, false, 0,
				CPDefinitionInventoryConstants.DEFAULT_MAX_ORDER_QUANTITY, null,
				1);
		}`

can you please have a look and fix the test?
Thank you!

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
@andrea-ale-sbarra Can you check if it's alright now? Thank you.

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java [hunk 1]
```diff
@@ -696,14 +696,14 @@ public void testLowStockActivityOnCheckout() throws Exception {
 		if (cpDefinitionInventory == null) {
 			_cpDefinitionInventoryLocalService.addCPDefinitionInventory(
 				_user.getUserId(), cpDefinition.getCPDefinitionId(), "default",
-				"default", false, false, 1, false, 0,
+				"default", false, false, 1, false, 1,
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java:699]:
```suggestion
				"default", false, false, 1, false, CPDefinitionInventoryConstants.DEFAULT_MIN_ORDER_QUANTITY,
```

File: modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java [hunk 2]
```diff
@@ -696,14 +696,14 @@ public void testLowStockActivityOnCheckout() throws Exception {
...
 				CPDefinitionInventoryConstants.DEFAULT_MAX_ORDER_QUANTITY, null,
 				1);
 		}
 		else {
 			_cpDefinitionInventoryLocalService.updateCPDefinitionInventory(
 				cpDefinitionInventory.getCPDefinitionInventoryId(), "default",
-				"default", false, false, 1, false, 0,
+				"default", false, false, 1, false, 1,
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/test/CommerceCheckoutTest.java:706]:
```suggestion
				"default", false, false, 1, false, CPDefinitionInventoryConstants.DEFAULT_MIN_ORDER_QUANTITY,
```

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by victorcdp** [comment]
@andrea-ale-sbarra Should be fine now. Thank you

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@victorcdp Andrea is on PTO, I will do the review. Can you move commits that contain auto-generated files at the end of the history? Thank you!

**Review by alessiorendina — APPROVED** [review]

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce

**Comment by luca-pellizzon** [comment]
@victorcdp 
modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/change/tracking/test/CPDefinitionInventoryTableReferenceDefinitionTest.java
must be updated.
The test is sending 0 to the minOrderQuantity param.
Check line 74 and update it so that this PR will pass CI tests.
Thanks

cc @brianikim

**Comment by victorcdp** [comment]
@luca-pellizzon Should be fine now. Thank you.

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
Thanks @victorcdp and @luca-pellizzon for the update.

QA passed. It will be resent after writing the automation test.

**Comment by jaydawu** [comment]
Automation finished and resent on https://github.com/liferay-commerce/liferay-portal/pull/3633

**Comment by jaydawu** [comment]
ci:reopen

**Comment by jaydawu** [comment]
Hi @victorcdp, could you help solve the semantic versioning issue [here](https://testray.liferay.com/home/-/testray/case_results/2260564198?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2260557136%26testrayRunId%3D2260560902%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc), thanks!

**Comment by brianikim** [comment]
ci:close

**Pull request closed without merging**